lib/octokit/repository.rb

```diff
@@ -71,5 +78,17 @@ def url
     alias :user :owner
     alias :username :owner
     alias :repo :name
+
+    private
+
+      def validate_owner_and_name!
+        if @owner.match(/\//) || @name.match(/\//) || !url.match(/\A#{URI.regexp}\z/)
```

Comment:
Let's use [`include`](http://ruby-doc.org/core-1.9.3/String.html#method-i-include-3F) here for the first two checks.